src/utils/nanvixd/src/sandbox/microvm.rs

```diff
@@ -119,19 +120,38 @@ impl Microvm {
         // After the user VM has started, accept the incoming connection for the control-plane.
         // Post-condition: once the connection has been accepted, the user VM has been able to
         // connect to the system VM (if an address is provided).
-        let control_plane_stream: SocketStream = match control_plane_listener.accept_timeout(
-            control_plane_poll,
+        let control_plane_stream: SocketStream = match timeout(
             Duration::from_secs(config::syscomm::ACCEPT_TIMEOUT_SECS),
-        ) {
-            Ok(stream) => stream,
-            Err(e) => {
+            ::syscomm::r#async::accept(control_plane_listener),
+        )
+        .await
+        {
+            Ok(Ok(stream)) => stream,
+            Ok(Err(e)) => {
                 // If the user VM has not accepted the control-plane connection, it means that
                 // something went wrong during start-up. We kill the process ignoring errors,
                 // and return an error.
                 let reason: String =
                     format!("error connecting control-plane to user VM (error={e:?})");
                 error!("{reason}");
 
+                // FIXME: use send_sigkill_to_child after it lands
```

Comment:
I add a `Self::send_sigkill_to_child` in #1033, so, if #1033 lands first, I will remove this `FIXME` and use the method introduced there.